layer/decorators/layer_wrapper.py

```diff
@@ -62,3 +65,44 @@ def __init__(
                         "Dependencies can only be a string, Dataset or Model."
                     )
         self.layer.set_dependencies(paths)
+
+    def get_definition(self) -> FunctionDefinition:
+        self.layer.validate()
+        current_project_full_name = get_current_project_full_name()
+
+        # get asset dependencies
+        asset_dependencies: List[AssetPath] = []
+        for d in self.layer.get_dependencies():
+            full_path_dep = d
+            if d.is_relative():
+                if d.project_name is not None:
+                    full_path_dep = d.with_project_full_name(
+                        ProjectFullName(
+                            account_name=current_project_full_name.account_name,
+                            project_name=d.project_name,
+                        )
+                    )
+                else:
+                    full_path_dep = d.with_project_full_name(current_project_full_name)
+            asset_dependencies.append(full_path_dep)
```

Comment:
nit: I'd put this into a separate method. same for get_pip_dependencies below

Reply:
done